frontend/packages/ux-editor/src/hooks/mutations/useAddOptionMutation.ts

```diff
@@ -0,0 +1,16 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useServicesContext } from 'app-shared/contexts/ServicesContext';
+import { QueryKey } from 'app-shared/types/QueryKey';
+
+export const useAddOptionMutation = (org: string, app: string) => {
+  const queryClient = useQueryClient();
+  const { uploadOption } = useServicesContext();
+
+  return useMutation({
+    mutationFn: (payload: FormData) => uploadOption(org, app, payload),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: [QueryKey.OptionListIds, org, app] }).then();
+      queryClient.invalidateQueries({ queryKey: [QueryKey.OptionLists, org, app] }).then();
+    },
```

Comment:
`onSuccess` støtter asynkrone funskjoner. Ved å bruke det, slipper du å "jukse" med `.then()`. Du kan også bruke `Promise.all` for at ikke den andre funskjonen skal vente på den første. 
```suggestion
    onSuccess: async () => {
      await Promise.all([
        queryClient.invalidateQueries({ queryKey: [QueryKey.OptionListIds, org, app] }),
        queryClient.invalidateQueries({ queryKey: [QueryKey.OptionLists, org, app] }),
      ]);
    },
```